src/NHibernate/Dialect/Function/CastFunction.cs

```diff
@@ -87,14 +87,7 @@ public SqlString Render(IList args, ISessionFactoryImplementor factory)
 				throw new QueryException(string.Format("invalid Hibernate type for cast(): type {0} not found", typeName));
 			}
 
-			if (CastingIsRequired(sqlType))
-			{
-				return new SqlString("cast(", args[0], " as ", sqlType, ")");
-			}
-			else
-			{
-				return new SqlString("(", args[0], ")");
-			}
+			return CastingIsRequired(sqlType) ? Render(args, sqlType, factory) : new SqlString("(", args[0], ")");
```

Comment:
With Render being virtual, you should remove CastingIsRequired and override Render in the TransparentCastFunction class to avoid this check.

Reply:
Nevermind, I didn't see that SQLiteCastFunction overrides CastingIsRequired as well and returns true/false based on the sqlType. I think these changes are probably fine.

Reply:
The SQLiteCastFunction class is slated to be removed after v5.3 so maybe the refactoring I suggest above should be done at that time as well.

Reply:
Made `CastingIsRequired` obsolete and overridden the `Render` method in `TransparentCastFunction`.